tests/components/thethingsnetwork/test_config_flow.py

```diff
@@ -0,0 +1,116 @@
+"""Define tests for the The Things Network onfig flows."""
+
+from ttn_client import TTNAuthError
+
+from homeassistant.components.thethingsnetwork.const import (
+    CONF_API_KEY,
+    CONF_APP_ID,
+    CONF_HOSTNAME,
+    DOMAIN,
+    TTN_API_HOSTNAME,
+)
+from homeassistant.config_entries import SOURCE_REAUTH, SOURCE_USER
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from . import add_schema_suggestion
+from .conftest import API_KEY, APP_ID, CONFIG_ENTRY, HOSTNAME
+
+USER_DATA = {CONF_HOSTNAME: HOSTNAME, CONF_APP_ID: APP_ID, CONF_API_KEY: API_KEY}
+USER_DATA_PARTIAL = {CONF_APP_ID: APP_ID, CONF_API_KEY: API_KEY}
+
+
+async def test_user(
+    hass: HomeAssistant, mock_TTNClient_coordinator, mock_TTNClient_config_flow
+) -> None:
+    """Test user config."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+    )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    schema = result["data_schema"]
+    user_data = schema(add_schema_suggestion(schema.schema, USER_DATA_PARTIAL))
+    assert user_data[CONF_HOSTNAME] == TTN_API_HOSTNAME  # Default value
```

Comment:
Why do we do this?

Reply:
to ensure the default hostname is set by the config_flow

Reply:
but why the whole `schema(add_schema_suggestion`?

Reply:
since hostname is a suggested value, the add_schema_suggestion is used to expand it into user_data. This emulates what the UI does.

The objective is to test that the suggested value is present in the schema. Is there any other better way you would propose?

Reply:
I don't think we do this anywhere, so I am a bit in the direction of "lets just remove it"